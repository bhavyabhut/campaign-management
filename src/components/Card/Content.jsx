import React from 'react';
import PropTypes from 'prop-types';
import { Tag } from 'antd';

import style from './content.module.css';
import tagColor from '../../constants/colors';
import { timeDifference } from '../../utility/helpers/general';

const Content = ({ data }) => (
  <>
    <div className={style.mrb}>
      <span className={style.customLabel}>Creator Name: </span>
      <span>{data.creator_name}</span>
    </div>
    <div className={style.mrb}>
      <span className={style.customLabel}>Status: </span>
      <span style={{ color: `${data.status === 1 ? 'green' : 'yellow'}` }}>
        {data.status === 1 ? 'Start' : 'Paused'}
      </span>
    </div>
    <div className={style.mrb}>
      <span className={style.customLabel}>Tags: </span>
      {data.tags &&
        data.tags.map(tagData => {
          const num = tagData.charCodeAt(0) + tagData.charCodeAt(tagData.length - 1);
          return (
            <Tag key={tagData} style={{ color: 'black' }} color={tagColor[num % 11]}>
              {tagData}
            </Tag>
          );
        })}
    </div>
    <div className={style.mrb}>
      <span className={style.customLabel}>Created Time: </span>
      <span>{timeDifference(data.created_at)}</span>
    </div>
    <div className={style.mrb}>
      <span className={style.customLabel}>Created At: </span>
      <span>{data.created_at}</span>
    </div>
    <div className={style.mrb}>
      <span className={style.customLabel}>Updated Time: </span>
      <span>{timeDifference(data.updated_at)}</span>
    </div>
    <div className={style.mrb}>
      <span className={style.customLabel}>Updated At: </span>
      <span>{data.updated_at}</span>
    </div>
  </>
);

Content.propTypes = {
  data: PropTypes.instanceOf(Object).isRequired,
};

export default Content;
