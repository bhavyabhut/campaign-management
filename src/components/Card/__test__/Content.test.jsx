import React from 'react';
import ReactDom from 'react-dom';
import { render, cleanup } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import Content from '../Content';
import renderer from 'react-test-renderer';

const dummyData = [
  {
    creator_name: 'Bhavya',
    campaign_name: 'Email',
    status: 1,
    created_at: '',
    updated_at: '',
    tags: ['awesome', 'nice'],
  },
  {
    creator_name: 'John',
    campaign_name: 'Video call',
    status: 2,
    created_at: '',
    updated_at: '',
    tags: ['poor', 'bad'],
  },
];

afterEach(cleanup);

it('Renders without crashing', () => {
  const div = document.createElement('div');
  ReactDom.render(<Content data={{}}></Content>, div);
});

it('Renders Card Content Correctly', () => {
  const { container } = render(<Content data={dummyData[0]}></Content>);
  const test = document.createElement('div');
  test.classList.add('mrb');
  test.innerHTML = `<span class="customLabel">Creator Name: </span><span>Bhavya</span>`;
  expect(container.querySelector('div')).toEqual(test);
});

it('Renders Card Content Correctly for second ', () => {
  const { container } = render(<Content data={dummyData[1]}></Content>);
  const test = document.createElement('div');
  test.classList.add('mrb');
  test.innerHTML = `<span class="customLabel">Creator Name: </span><span>John</span>`;
  expect(container.querySelector('div')).toEqual(test);
});

it('Card Content Found match snapshot', () => {
  const tree = renderer.create(<Content data={dummyData[0]} />).toJSON();
  expect(tree).toMatchSnapshot();
});
