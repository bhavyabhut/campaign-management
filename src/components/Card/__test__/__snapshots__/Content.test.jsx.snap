// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card Content Found match snapshot 1`] = `
Array [
  <div
    className="mrb"
  >
    <span
      className="customLabel"
    >
      Creator Name: 
    </span>
    <span>
      Bhavya
    </span>
  </div>,
  <div
    className="mrb"
  >
    <span
      className="customLabel"
    >
      Status: 
    </span>
    <span
      style={
        Object {
          "color": "green",
        }
      }
    >
      Start
    </span>
  </div>,
  <div
    className="mrb"
  >
    <span
      className="customLabel"
    >
      Tags: 
    </span>
    <span
      className="ant-tag ant-tag-has-color"
      style={
        Object {
          "backgroundColor": "#ffa39e",
          "color": "black",
        }
      }
    >
      awesome
    </span>
    <span
      className="ant-tag ant-tag-has-color"
      style={
        Object {
          "backgroundColor": "#ffd591",
          "color": "black",
        }
      }
    >
      nice
    </span>
  </div>,
  <div
    className="mrb"
  >
    <span
      className="customLabel"
    >
      Created Time: 
    </span>
    <span>
      a few seconds ago
    </span>
  </div>,
  <div
    className="mrb"
  >
    <span
      className="customLabel"
    >
      Created At: 
    </span>
    <span>
      
    </span>
  </div>,
  <div
    className="mrb"
  >
    <span
      className="customLabel"
    >
      Updated Time: 
    </span>
    <span>
      a few seconds ago
    </span>
  </div>,
  <div
    className="mrb"
  >
    <span
      className="customLabel"
    >
      Updated At: 
    </span>
    <span>
      
    </span>
  </div>,
]
`;
